tests/test_custom_checkers/test_invalid_range_index_checker.py

```diff
@@ -1,179 +1,66 @@
-                msg_id="invalid-range-index",
-                node=range_node,
-                args="2",
-            ),
-            ignore_position=True,
-        ):
-            self.checker.visit_call(range_node)
-
-    def test_invalid_range_two_arg(self):
-        src = """
-        range(5, 6)
-        """
+    @pytest.mark.parametrize(
+        "src, expected_arg",
+        [
+            ("range()", "1"),
+            ("range(-10)", "1"),
+            ("range(5, 6)", "1"),
+            ("range(2, 15, 20)", "1"),
+            ("range(1, 10, 0)", "1"),
+            ('range("hello", "bye")', "1"),
+        ],
+    )
+    def test_invalid_ranges(self, src, expected_arg):
         range_node = astroid.extract_node(src)
         with self.assertAddsMessages(
             pylint.testutils.MessageTest(
                 msg_id="invalid-range-index",
                 node=range_node,
-                args="2",
+                args=expected_arg,
             ),
             ignore_position=True,
         ):
             self.checker.visit_call(range_node)
 
-    def test_invalid_range_three_arg(self):
-        src = """
-        range(2, 15, 20)
-        """
-        range_node = astroid.extract_node(src)
-        with self.assertAddsMessages(
-            pylint.testutils.MessageTest(
-                msg_id="invalid-range-index",
-                node=range_node,
-                args="2",
-            ),
-            ignore_position=True,
-        ):
-            self.checker.visit_call(range_node)
-
-    def test_invalid_range_zero_step(self):
-        src = """
-        range(1, 10, 0)
-        """
-        range_node = astroid.extract_node(src)
-        with self.assertAddsMessages(
-            pylint.testutils.MessageTest(
-                msg_id="invalid-range-index",
-                node=range_node,
-                args="2",
-            ),
-            ignore_position=True,
-        ):
-            self.checker.visit_call(range_node)
-
-    def test_wrong_type(self):
-        src = """
-        range("hello", "bye")
-        """
-        range_node = astroid.extract_node(src)
-        with self.assertAddsMessages(
-            pylint.testutils.MessageTest(
-                msg_id="invalid-range-index",
-                node=range_node,
-                args="2",
-            ),
-            ignore_position=True,
-        ):
-            self.checker.visit_call(range_node)
-
-    def test_uninferable(self):
-        src = """
-        range(0, [][1])
-        """
-        range_node = astroid.extract_node(src)
-        with self.assertNoMessages():
-            self.checker.visit_call(range_node)
-
-    def test_variables_undefined(self):
-        src = """
-        range(start, stop)  # These variables are undefined
-        """
-        range_node = astroid.extract_node(src)
-        with self.assertNoMessages():
-            self.checker.visit_call(range_node)
-
-    def test_variables_defined(self):
```

Comment:
Leave this test and the test below as they are. They contain important comments that should not be lost when moving to parametrized tests.